test/Interop/C/struct/Inputs/noncopyable-struct.h

```diff
@@ -0,0 +1,19 @@
+// RUN: %target-typecheck-verify-swift -I %S/Inputs/
+
+typedef struct __attribute__((swift_attr("~Copyable"))) NonCopyable {
+  float x, y;
+} NonCopyable;
+
+typedef struct __attribute__((swift_attr("~Copyable"))) __attribute__((swift_attr("destroy:freeNonCopyableWithDeinit"))) NonCopyableWithDeinit {
```

Comment:
Do we plan to support `destroy:` with copyable types? If not, should we diagnose when people try to add it to a copyable type?

Maybe we should have a diagnostic when people try to add it to a C++ type that already has a non-trivial dtor.

Reply:
Copyable value types in Swift don't support user-defined `deinit`, so we shouldn't try to support it for C structs, either.

Reply:
I don't know what to do about C++ types a non-trivial destructor. Warn about and ignore the `destroy:` operation, perhaps?

Reply:
> Warn about and ignore the destroy: operation, perhaps?

I think that makes sense.

Reply:
I've done that in the latest commit